bokehjs/src/coffee/models/glyphs/webgl/markers.coffee

```diff
@@ -0,0 +1,524 @@
+    attribute vec4  a_bg_color;
+    //
+    varying float v_linewidth;
+    varying float v_size;
+    varying vec4  v_fg_color;
+    varying vec4  v_bg_color;
+    varying vec2  v_rotation;
+
+    void main (void)
+    {
+        v_size = a_size;
+        v_linewidth = a_linewidth;
+        v_fg_color = a_fg_color;
+        v_bg_color = a_bg_color;
+        v_rotation = vec2(cos(-a_angle), sin(-a_angle));
+        // Calculate position - the -0.5 is to correct for canvas origin
+        vec2 pos = (vec2(a_x, a_y) + u_offset) * u_scale - vec2(0.5, 0.5); // in pixels
+        pos /= u_canvas_size;  // in 0..1
+        gl_Position = vec4(pos*2.0-1.0, 0.0, 1.0);
+        gl_Position.y *= -1.0;
+        gl_PointSize = SQRT_2 * v_size + 2.0 * (a_linewidth + 1.5*u_antialias);
+    }
+    """
+
+  FRAG: """
+    precision mediump float;
+    const float SQRT_2 = 1.4142135623730951;
+    const float PI = 3.14159265358979323846264;
+    //
+    uniform float u_antialias;
+    //
+    varying vec4  v_fg_color;
+    varying vec4  v_bg_color;
+    varying float v_linewidth;
+    varying float v_size;
+    varying vec2  v_rotation;
+
+    MARKERCODE
+
+    vec4 outline(float distance, float linewidth, float antialias, vec4 fg_color, vec4 bg_color)
+    {
+        vec4 frag_color;
+        float t = linewidth/2.0 - antialias;
+        float signed_distance = distance;
+        float border_distance = abs(signed_distance) - t;
+        float alpha = border_distance/antialias;
+        alpha = exp(-alpha*alpha);
+
+        // If fg alpha is zero, it probably means no outline. To avoid a dark outline
+        // shining through due to aa, we set the fg color to the bg color. Avoid if (i.e. branching).
+        float select = float(bool(fg_color.a));
+        fg_color.rgb = select * fg_color.rgb + (1.0  - select) * bg_color.rgb;
+        // Similarly, if we want a transparent bg
+        select = float(bool(bg_color.a));
+        bg_color.rgb = select * bg_color.rgb + (1.0  - select) * fg_color.rgb;
+
+        if( border_distance < 0.0)
+            frag_color = fg_color;
+        else if( signed_distance < 0.0 ) {
+            frag_color = mix(bg_color, fg_color, sqrt(alpha));
+        } else {
+            if( abs(signed_distance) < (linewidth/2.0 + antialias) ) {
+                frag_color = vec4(fg_color.rgb, fg_color.a * alpha);
+            } else {
+                discard;
+            }
+        }
+        return frag_color;
+    }
+
+    void main()
+    {
+        vec2 P = gl_PointCoord.xy - vec2(0.5, 0.5);
+        P = vec2(v_rotation.x*P.x - v_rotation.y*P.y,
+                 v_rotation.y*P.x + v_rotation.x*P.y);
+        float point_size = SQRT_2*v_size  + 2.0 * (v_linewidth + 1.5*u_antialias);
+        float distance = marker(P*point_size, v_size);
+        gl_FragColor = outline(distance, v_linewidth, u_antialias, v_fg_color, v_bg_color);
+        //gl_FragColor.rgb *= gl_FragColor.a;  // pre-multiply alpha
+    }
+    """
```

Comment:
It should be fairly easy. There are gulp extensions for templating source files. The only pain point is mocha, which bypasses the build and compiles sources by itself. We have to change that, because I already had issues with getting typescript to work in bokehjs tests (it doesn't currently).